nested-types.ipynb

```diff
@@ -0,0 +1,802 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Nested types\n",
+    "\n",
+    "* List, List View\n",
+    "* Struct\n",
+    "* Map\n",
+    "* Union\n",
+    "* Dictionary encoded,\n",
+    "* Run-end encoded\n",
+    "\n",
+    "We will have a look at first three types."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Fixed and variable size list\n",
+    "\n",
+    "- **Fixed size list:** each column slot contains a fixed size sequence of values all having the same type\n",
+    "- **Variable size list:** column slot sequence size is not defined/set\n",
+    "- List has an offsets of 32-bits and **large** list has 64-bit offsets\n",
+    "\n",
+    "![image info](./diagrams/fixed-list-diagram.svg)\n",
+    "![image info](./diagrams/var-list-diagram.svg)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import nanoarrow as na\n",
+    "import numpy as np\n",
+    "import pyarrow as pa"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Variable size list"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<pyarrow.lib.ListArray object at 0x110840ca0>\n",
+       "[\n",
+       "  [\n",
+       "    12,\n",
+       "    -7,\n",
+       "    25\n",
+       "  ],\n",
+       "  null,\n",
+       "  [\n",
+       "    0,\n",
+       "    -127,\n",
+       "    127,\n",
+       "    50\n",
+       "  ],\n",
+       "  []\n",
+       "]"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "column_1 = pa.array([[12, -7, 25], None, [0, -127, 127, 50], []],\n",
+    "                    type=pa.list_(pa.int8()))\n",
+    "column_1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[<pyarrow.Buffer address=0x4bdc40200c0 size=1 is_cpu=True is_mutable=True>,\n",
+       " <pyarrow.Buffer address=0x4bdc4020080 size=20 is_cpu=True is_mutable=True>,\n",
+       " None,\n",
+       " <pyarrow.Buffer address=0x4bdc4020100 size=7 is_cpu=True is_mutable=True>]"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Inspecting buffers using PyArrow and buffers() method\n",
```

Comment:
Note here that this returns all buffers of both the list array itself and its child arrays